### ISSUE TITLE
Setting custom anonymous ID

### ISSUE THREAD

**Issue by bartschelfhout** [segmentio/consent-manager#101]
We are using our own anonymous ID with segment in order to reduce the number of different user identifiers in our data space.

We achieve this by adding the following piece of code prior to any tracking code firing;

analytics.ready(function() {
    analytics.user().anonymousId(our_anonymous_identifier);
});

When using the consent manager, analytics is not ready until user interaction with the banner, which consequently fires an identify call off to the segment end point. This means that this identify call will contain a segment client side generated anonymous ID.

question:
Is there any future option to make sure we can set the anonymous ID in the consent manager on time of loading?

Thanks

**Comment by alexhansonsmith** [comment]
Hi @bartschelfhout - did you get anywhere with this? Without being able to control the identifier our MTU usage has spiked considerably

**Comment by bartschelfhout** [comment]
Hi @alexhansonsmith I did actually.
Just to give some context fist:
Our own anonymous identifier is being created by a piece of javascript that is loaded from the head of our website, hence the identifier is available in way before the entirety of the page loads.

What we discovered is that the consent manager does a series of lookups before generating a new identifier, to find if there was a previously generated one. 
It looks at the local storage for a variable called `ajs_anonymous_id`. 

So the solution was simple. In the piece of javascript that is loaded from the header section of our page, we write our own anonymous identifier to localStorage with that name.
`localStorage && localStorage.setItem('ajs_anonymous_id', our_anonymous_identifier);`

This would always get picked up by the consent manager.
